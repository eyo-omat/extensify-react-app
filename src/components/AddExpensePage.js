import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startExpense} from "../actions/expenses";

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => ({startExpense: (expense) => dispatch(startExpense(expense))});


export default connect(undefined, mapDispatchToProps)(AddExpensePage);
