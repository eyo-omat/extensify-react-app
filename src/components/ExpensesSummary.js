import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';

const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formatExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formatExpenseTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link to="/create" className="button">Add Expense </Link>

                </div>
            </div>
        </div>

    );
};

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpense.length,
        expensesTotal: selectExpenseTotal(visibleExpense)

    };

};

export default connect(mapStateToProps)(ExpensesSummary);


