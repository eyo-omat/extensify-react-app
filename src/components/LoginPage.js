import React from 'react';
import {startLogin} from '../actions/auth';
import { connect } from 'react-redux';

export const LoginPage = ({ startLogin }) => {

    return (
        <div className="box-layout">

            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>Its time to get you expenses under control.</p>

                <button onClick={startLogin} className="button">Login with Google</button>
            </div>
        </div>

    );
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage)


