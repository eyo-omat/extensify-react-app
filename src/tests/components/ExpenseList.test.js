import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expsenseData from '../fixtures/expenses';

test('should render expenseList with expenses', () => {
   const wrapper = shallow(<ExpenseList expenses={expsenseData}/>);
   expect(wrapper).toMatchSnapshot();

});

test('should render expenseList with empty message', () => {
   const wrapper = shallow(<ExpenseList expenses={[]}/>);
   expect(wrapper).toMatchSnapshot();

});
