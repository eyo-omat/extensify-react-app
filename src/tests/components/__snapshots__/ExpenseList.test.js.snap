// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList with empty message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <p>
        No expenses
      </p>
    </div>
  </div>
</div>
`;

exports[`should render expenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={4500}
      createdAt={0}
      description="A new description"
      id="1"
      key="1"
      note="A new note"
    />
    <ExpenseListItem
      amount={450}
      createdAt={345600000}
      description="Mac"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={3}
      createdAt={2073600000}
      description="Credit card"
      id="3"
      key="3"
      note="visa"
    />
    <ExpenseListItem
      amount={5}
      createdAt={-345600000}
      description="Gum"
      id="4"
      key="4"
      note="Chewy"
    />
  </div>
</div>
`;
